### ISSUE TITLE
False unsubscriptable-object positive with generic subprocess.Popen and python3.9

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#4034]
### Steps to reproduce

```shellsession
$ cat /tmp/t.py
from subprocess import Popen, PIPE

def foo(popen: Popen[str]) -> None:
    pass

foo(Popen(["ls"], stdout=PIPE))
```

### Current behavior

```shellsession
$ pylint /tmp/t.py | grep subs
/tmp/t.py:3:15: E1136: Value 'Popen' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No `unsubscriptable-object` error.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.0 (default, Dec 13 2020, 12:54:06) 
[GCC 7.5.0]
```

**Comment by dbaty** [comment]
tl;dr: on Python 3.8 and earlier, everything seems fine, I don't think there is a bug. With Python 3.9, though, this is a bug and a fix has been proposed in Astroid (a dependency of pylint): PyCQA/astroid/pull/882.

Maintainers: I would mention "under Python 3.9" in the title and add the corresponding issue tag. :)


#### On Python 3.8 and earlier

`Popen` is indeed **not** subscriptable with in Python 3.8 and earlier. For example, running your code with Python 3.8 I get:

```
$ python3.8 test.py
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    def foo(popen: Popen[str]) -> None:
TypeError: 'type' object is not subscriptable
```

So I think that pylint is right to report an error here **up to Python 3.8**.

#### On Python 3.9

However, under Python 3.9, `Popen` **is** subscriptable. Running your snippet with Python 3.9 does not raise any error.

So, with Python 3.9, pylint should **not** report the `unsubscriptable-object` warning. Yet it does, which is a bug.

To determine whether a class is subscriptable, pylint looks for a `__class_getitem__` method. Here it cannot find one. This is because Astroid "overrides" the definition of `subprocess.Popen` (with what Astroid calls a "brain") and there the `__class_getitem__` method is missing.

I proposed a fix in Astroid: PyCQA/astroid/pull/882.

**Comment by scop** [comment]
Oh, interesting. The error was provoked by a fix to an issue flagged by running mypy with python 3.8 in python_version 3.8 mode. I think that makes it a mypy (well or typeshed) bug, too. Thanks!

**Comment by scop** [comment]
Ref https://github.com/python/typeshed/issues/4948

**Comment by hippo91** [comment]
@scop thanks for the report. @dbaty thanks for your investigation and corresponding PR.
